src/odemis/gui/model/__init__.py

```diff
@@ -1433,24 +1402,47 @@ def __init__(self, microscope):
         # POS_ACTIVE_RANGE contains the bounding-box of the positions with a sample
         carrier_range = md[model.MD_POS_ACTIVE_RANGE]
         minx, miny = carrier_range["x"][0], carrier_range["y"][0]  # bottom-left of carrier 1 in m
-
-        # TODO: in the future, there could be an additional argument in the configuration file to specify
-        #  the parameters of the sample carrier. For now, only one design is supported and hardcoded.
-        carrier_params = self.SAMPLE_CARRIER_3x3
+        # SAMPLE_CENTERS contains the center position of the scintillators from bottom-left
+        # position with a sample
+        if model.MD_SAMPLE_CENTERS not in md:
+            raise KeyError("Stage has no MD_SAMPLE_CENTERS metadata.")
+        centers = md[model.MD_SAMPLE_CENTERS]
+        layout = sample_positions_to_layout(centers)
+        for row_idx, row in enumerate(layout):
+            for column_idx, name in enumerate(row):
+                scintillator_number = int(name.split()[-1])
```

Comment:
In case `int()` fails, please give a nice error message indicating that the name of the sample must have a number in second place.